iroh-net/src/magicsock/metrics.rs

```diff
@@ -90,6 +92,7 @@ impl Default for Metrics {
             recv_data_ipv4: Counter::new("recv_data_ipv4"),
             recv_data_ipv6: Counter::new("recv_data_ipv6"),
             recv_datagrams: Counter::new("recv_datagrams"),
+            recv_gro_packets: Counter::new("recv_gro_packets"),
```

Comment:
I'd prefer to call this "datagram", so `recv_gro_datagrams`.

I think this is in line with the RFC9000 terminology:
- A datagram is a single UDP, err... datagram sent.  This is pretty universal UDP networking terminology.
- A packet is something that can be sent in a single datagram.  Though some packets can be coalesced together into a datagram where needed (mostly the handshake).
- A frame is what is put in packets, again you can have multiple frames per packet.

Reply:
yes, our log messages confuse this terminology too.  one day that will be cleaned up too :)

Reply:
Yeah I went back and forth on that quite a bit.

Generally love any advice to consistent naming here. I've spent some time googling around, but people are arguing many ways.

You say "some packets can be coalesced together into a datagram", what kinds of packets are we talking about here?
This is super confusing to my ears because when I hear "packet" I think "IP packet". But I guess it can also mean "QUIC Packet"?